pkg/middlewares/metrics/metrics.go

```diff
@@ -94,7 +96,26 @@ func (m *metricsMiddleware) ServeHTTP(rw http.ResponseWriter, req *http.Request)
 
 	labels = append(labels, "code", strconv.Itoa(recorder.getCode()))
 	m.reqsCounter.With(labels...).Add(1)
-	m.reqDurationHistogram.With(labels...).Observe(time.Since(start).Seconds())
+	m.reqDurationHistogram.With(labels...).Observe(m.getDurationValue(time.Since(start)))
+}
+
+func (m *metricsMiddleware) getDurationValue(d time.Duration) float64 {
+	switch m.timeScale {
```

Comment:
This whole switch does not work if someone creates a standardRegistry with e.g. `timeScale = 500 * time.Millisecond`, right?
Shouldn't we do something about that?
At the very least, I think the default case L116 should be used to catch the above case, and return an error or generate a warning.

Reply:
yes, Initially I thought it would be better to just fallback to Seconds (current behavior) if any other value was passed rather than introducing a new error, but that can lead to some tricky situations in the future. I'll add the error return.

Reply:
I just noted that there is another problem in this PR and it won't work as expected on `metrics.NewMultiRegistry`. Fixing this before I update again.